plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -254,40 +310,44 @@ export class PgInstrumentation extends InstrumentationBase {
           }
         }
 
-        const connectResult: unknown = originalConnect.call(
-          this,
-          callback as never
+        const connectResult: unknown = context.with(
+          trace.setSpan(context.active(), span),
+          () => {
+            return originalConnect.call(this, callback as never);
+          }
         );
 
-        // No callback was provided, return a promise instead
-        if (connectResult instanceof Promise) {
-          const connectResultPromise = connectResult as Promise<unknown>;
-          return context.bind(
-            context.active(),
-            connectResultPromise
-              .then(result => {
-                // Return a pass-along promise which ends the span and then goes to user's orig resolvers
-                return new Promise(resolve => {
-                  span.end();
-                  resolve(result);
-                });
-              })
-              .catch((error: Error) => {
-                return new Promise((_, reject) => {
-                  span.setStatus({
-                    code: SpanStatusCode.ERROR,
-                    message: error.message,
-                  });
-                  span.end();
-                  reject(error);
-                });
-              })
-          );
-        }
-
-        // Else a callback was provided, so just return the result
-        return connectResult;
+        return handleConnectResult(span, connectResult);
       };
     };
   }
 }
+
+function handleConnectResult(span: Span, connectResult: unknown) {
+  if (!(connectResult instanceof Promise)) {
+    return connectResult;
+  }
+
+  const connectResultPromise = connectResult as Promise<unknown>;
+  return context.bind(
+    context.active(),
+    connectResultPromise
+      .then(
+        result =>
+          new Promise(resolve => {
+            span.end();
+            resolve(result);
+          })
+      )
+      .catch((error: Error) => {
+        return new Promise((_, reject) => {
+          span.setStatus({
+            code: SpanStatusCode.ERROR,
+            message: error.message,
+          });
+          span.end();
+          reject(error);
+        });
```

Comment:
No need to create Promises using the constructor here. See https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1125#discussion_r967104209

Reply:
Thanks, fixed